[Artigo] How I Start: Elixir

rhiroyuki opened the pull request:
Tradução sendo realizada junto com o @caiquemitsuoka 
para mais informações veja a issue #43

CaiqueMitsuoka committed:
Adiciona elixir/como-comecar

rhiroyuki committed:
Adiciona introducao do artigo

rhiroyuki committed:
Adiciona secao de pattern matching

rhiroyuki committed:
Adiciona secao transferencia com portais

rhiroyuki committed:
Adiciona traducao da secao insp. portais com protocolos

rhiroyuki committed:
Adiciona secao transferencias distribuidas

felipesoares6 commented:
@CaiqueMitsuoka @rhiroyuki @JulioFortunato como está esse artigo, pretendem continuar a tradução?

juunegreiros commented:
Eai @CaiqueMitsuoka e @rhiroyuki, beleza? Como está o andamento dessa tradução? Consigo ajudar a traduzir?

rhiroyuki commented:
Olá @juunegreiros, pode sim. Falta pouca coisa. Vou mandar o artigo original para você dar uma lida.
http://howistart.org/posts/elixir/1/

Você pode traduzir as partes em que está escrito 'Caique'.

Obrigado pela ajuda :)

juunegreiros committed:
Add Instalação

juunegreiros committed:
Add Nosso Primeiro Projeto

juunegreiros committed:
Add Modelando portais com agentes

juunegreiros committed:
Add Atirando em acessos supervisionados

juunegreiros committed:
Adiciona finalizando

juunegreiros committed:
Terminando a tradução

Estiliza o texto com markdown, traduz os comentários nos códigos e altera alguns detalhes para que o texto possa fazer mais sentido.

rhiroyuki commented on elixir/como-comecar.md:
````diff
@@ -81,9 +165,109 @@ iex> [0|list]
+```elixir
+defmodule Portal.Acesso do
+  @doc """
+  Inicia uma dor com a `cor` recebida.
````
No `dor` era pra ser `acesso`?

juunegreiros replied:
Era! Porque que eu coloquei dor eu não sei não uahauhauha Arrumarei

juliofortunato commented on elixir/como-comecar.md:
```diff
@@ -0,0 +1,631 @@
+
+Excelente, nós já temos um projeto rodando com um ambiente de testes configurado.
+
+Vamos explorar o projeto gerado usando um editor de texto. Eu pessoalmente não presto muita atenção em editores de texto, normalmente uso o [Sublime Text 3](http://www.sublimetext.com/3) mas você pode achar [suporte Elixir para diferentes editores de texto no site](http://elixir-lang.org/) na sessão "Code Editor Support".
```
Achei essa parte um pouco confusa.
O que acham de mudar:

> você pode achar suporte Elixir para diferentes editores de texto no site na sessão "Code Editor Support"

para:

> "você pode encontrar diferentes editores de texto com suporte ao Elixir no site na sessão "Code Editor Support"

?

juliofortunato commented on elixir/como-comecar.md:
```diff
@@ -0,0 +1,631 @@
+
+# Pattern matching
+
+Antes de implementarmos nosso aplicativo, precisamos falar sobre pattern matching. O operador = no Elixir é um pouco diferente do que vemos em outros idiomas:
```
Aqui eu trocaria `idiomas` por `linguagens`, para se adequar melhor ao contexto de desenvolvimento de software

juliofortunato commented on elixir/como-comecar.md:
```diff
@@ -0,0 +1,631 @@
+
+## Modelando portais com agentes
+
+As estruturas de data do Elixir são imutáveis. Nos exemplos acima, nós nunca mudamos a lista. Nós podemos quebrar em uma lista à parte ou adicionar novos elementos na cabeça, mas a lista original nunca é modificada.
```
"As estruturas de **dados** do Elixir [...]"

juliofortunato commented on elixir/como-comecar.md:
```diff
@@ -0,0 +1,631 @@
+  @doc """
+  Remove um valor do `acesso`
+
+  Returna `{:ok, valor}` se existe um valor
```
**Retorna** {:ok, valor} [...]

juliofortunato commented on elixir/como-comecar.md:
```diff
@@ -0,0 +1,631 @@
+{:ok, #PID<0.74.0>}
+
+# Inicia transferência da lista [1, 2, 3, 4] do laranja para o azul
+iex(3)> portal = Portal.transfere(:laranja, :azul, [1, 2, 3, 4])
```
Aqui estamos dando nomes em português para as funções como `Portal.trasfere()`, `Portal.atirar()` e `Portal.inserir_direita()`, mas nos exemplos de código ao longo do artigo, estão sendo implementada em inglês.

juliofortunato commented on elixir/como-comecar.md:
```diff
@@ -0,0 +1,631 @@
+Starts transfering `data` from `left` to `right`.
+"""
+def transfer(left, right, data) do
+  # First add all data to the portal on the left
```
Como nos outros trechos de código os comentários foram traduzidos, eu sugiro manter esse padrão e traduzir aqui também.

juliofortunato commented on elixir/como-comecar.md:
````diff
@@ -0,0 +1,631 @@
+    (portal) lib/portal.ex:25: Portal.push_right/1
+```
+
+Nós recebemos uma mensagem de erro porque a porta `:azul` não existe. Você pode ver a mensagem `** (EXIT) no process` depois da nossa chamada da função. Para concertar a situação, nós vamos configurar um supervisor que será responsável por reiniciar um acesso do portal sempre que ele quebrar.
````
"concertar"  -> **consertar**

felipesoares6 commented on elixir/como-comecar.md:
```diff
@@ -614,18 +614,19 @@ Nossa transferência de portal distribuído funciona porque as portas são apena
-  - Aprenda mais sobre [ExUnit](http://elixir-lang.org/docs/stable/ex_unit/ExUnit.html), o framework de test do Elixir, e escreva testes para a funcionalidade que acabamos de construir. Lembre que já fizemos uma estrutura padrão estabelecida no diretório `test`.
-  - Gere a documentação HTML para o seu projeto com [ExDoc](http://github.com/elixir-lang/ex_doc)
-  - Suba o seu projeto em um repositório externo, como [Github](https://github.com/), e publique um pacote usando o [Hex package manager](https://hex.pm/).
+* Adicione uma funcão `Portal.push_left/1` que transfira os dados na outra direção. Como você pode evitar o código duplicado existente entre as funções `push_left/1` e `push_right/1`?
```
função *

felipesoares6 commented on elixir/como-comecar.md:
```diff
@@ -614,18 +614,19 @@ Nossa transferência de portal distribuído funciona porque as portas são apena
+ * Suba o seu projeto em um repositório externo, como [Github](https://github.com/), e publique um pacote usando o [Hex package manager](https://hex.pm/).
 
-Nós convidamos você para explorar nosso [site](http://elixir-lang.org/) e ler nosso guia Getting Started ou muitos dos recursos disponíveis para aprender mas sobre Elixir e nossa comunidade vibrante.
+Nós convidamos você para explorar nosso [site](http://elixir-lang.org/) e ler nosso guia Getting Started ou outros dos muitos recursos disponíveis para aprender mas sobre Elixir e nossa comunidade vibrante.
```
aprender mais*

felipesoares6 commented on elixir/como-comecar.md:
```diff
@@ -614,18 +614,19 @@ Nossa transferência de portal distribuído funciona porque as portas são apena
 Nos vemos por aí!
 
 José Valim
+m
```
:fire:

felipesoares6 commented:
@juunegreiros Poderia adicionar ao readme o novo tópico Elixir e listar esse artigo lá? :smile:
